### PR TITLE
CMR-4495 variable force delete

### PR DESCRIPTION
**Pull request by oubiwann**
Updated cascading delete behaviour for variables to match that of services.

### CODE REVIEW

File: metadata-db-app/int_test/cmr/metadata_db/int_test/concepts/force_delete_test.clj [hunk 1]
```diff
@@ -124,3 +128,52 @@
...
+        var-assn-concept-id (:concept-id var-assn)]
+    (testing "initial conditions"
+      ;; creation results as expected
+      (is (= 3 (:revision-id var-concept)))
+      ;; variable revisions in db
+      (is (= 3
+             (count (:concepts (util/find-concepts :variable)))))
+      (is (= 200 (:status (util/get-concept-by-id-and-revision
+                           var-concept-id 3))))
+      (is (= 200 (:status (util/get-concept-by-id-and-revision
```
Comment (ygliuvt) [inline metadata-db-app/int_test/cmr/metadata_db/int_test/concepts/force_delete_test.clj]: For completeness, add checking of revision 1 exists in all the testing cases.

File: system-int-test/test/cmr/system_int_test/search/variable/force_delete_test.clj [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      ;; make sure that the collection's `associations` field is present, that
+      ;; it contains `:variables` and that the right variable is associated
+      (is (= coll-concept-id
+             (get-in assn-response [:associated-item :concept-id])))
+      (is (contains? (get-collection-variables) var-concept-id)))
+    (testing "just the last revision is deleted"
+      (mdb/force-delete-concept var-concept-id 2)
+      ;; make sure the variable association has not been deleted
+      (is (contains? (get-collection-variables) var-concept-id)))
+    (testing "just the most recent revision is deleted"
```
Comment (ygliuvt) [inline system-int-test/test/cmr/system_int_test/search/variable/force_delete_test.clj]: We should add back the variables all revisions search test that is removed from `variable_revisions_search_test.clj` when you move the force delete tests here.

File: metadata-db-app/int_test/cmr/metadata_db/int_test/concepts/force_delete_test.clj (hunk 1 above)
Comment (oubiwann) [reply metadata-db-app/int_test/cmr/metadata_db/int_test/concepts/force_delete_test.clj]: Done.

File: system-int-test/test/cmr/system_int_test/search/variable/force_delete_test.clj (hunk 2 above)
Comment (oubiwann) [reply system-int-test/test/cmr/system_int_test/search/variable/force_delete_test.clj]: Done.

File: system-int-test/test/cmr/system_int_test/search/variable/force_delete_test.clj [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+             (get-in assn-response [:associated-item :concept-id])))
+      (is (contains? (get-collection-variables) var-concept-id))
+      (variable/assert-variable-associations
+       var-concept {:collections [{:concept-id coll-concept-id}]}
+       {:all-revisions true}))
+    (testing "just the second-to-last revision is deleted"
+      (mdb/force-delete-concept var-concept-id 2)
+      ;; make sure the variable association has not been deleted
+      (is (contains? (get-collection-variables) var-concept-id))
+      ;; XXX the follwoing nil result is not what I would expect here ... is
```
Comment (oubiwann) [inline system-int-test/test/cmr/system_int_test/search/variable/force_delete_test.clj]: @ygliuvt and I are going to chat about the implications here ... he's been thinking along parallel lines.

**Comment by oubiwann** [comment]
I've just pushed up the latest changes and will check with Bamboo builds in the morning.

File: metadata-db-app/src/cmr/metadata_db/api/concepts.clj [hunk 4]
```diff
@@ -95,7 +95,10 @@
 (defn- force-delete
   "Permanently remove a concept version from the database."
   [context params concept-id revision-id]
-  (let [{:keys [revision-id]} (concept-service/force-delete context concept-id (as-int revision-id))]
+  (let [;; force? is only used for testing purpose, should not be used in real operations
```
Comment (oubiwann) [inline metadata-db-app/src/cmr/metadata_db/api/concepts.clj]: I think this is the most expedient way to do this right now ... but maybe in the future we can remove support for this and add a utility function in the tests that allows us to make direct, custom queries to the DB in order to induce inconsistencies against which we can test ...

**Pull request merged**